### PR TITLE
update_licensepool should handle unhelpful errors, and errors in building a series feed should be propagated

### PR DESCRIPTION
**Pull request by leonardr**
This branch fixes two unrelated issues: 

The overdrive collection reaper crash in https://github.com/NYPL-Simplified/circulation/issues/350

and https://github.com/NYPL-Simplified/circulation/issues/351

### CODE REVIEW

File: tests/test_overdrive.py [hunk 1]
```diff
@@ -323,12 +323,11 @@ def test_update_licensepool_error(self):
             "overdrive_availability_information.json"
         )
         api = DummyOverdriveAPI(self._db)
-        api.queue_response(response_code=500, content=availability)
+        api.queue_response(response_code=500, content="An error occured.")
```
Comment (leonardr) [inline tests/test_overdrive.py]: Our error handling only worked if you sent a non-200 response code but also provided the right data. That's the case we were testing but that's not actually how Overdrive deals with errors.

File: api/controller.py [hunk 2]
```diff
@@ -952,10 +952,16 @@ def series(self, series_name):
 
         lane = SeriesLane(self._db, series_name)
         annotator = self.manager.annotator(lane)
+        facets = load_facets_from_request()
+        if isinstance(facets, ProblemDetail):
+            return facets
+        pagination = load_pagination_from_request()
+        if isinstance(pagination, ProblemDetail):
+            return pagination
```
Comment (aslagle) [inline api/controller.py]: Cool, I had noticed this too. It's also a problem in the contributor lane.

Comment (aslagle) [inline api/controller.py]: https://github.com/NYPL-Simplified/circulation/issues/343

Comment (leonardr) [inline api/controller.py]: I'll take care of that one as well. Thanks for the link!

Comment (aslagle) [inline api/controller.py]: Also the reason we're seeing this is that the contributor and series feeds have links with pagination and facets that don't work - I made an issue for that too https://github.com/NYPL-Simplified/circulation/issues/344

**Comment by aslagle** [comment]
:+1:

File: api/controller.py (hunk 2 above)
Comment (leonardr) [inline api/controller.py]: Hmm, I won't fix that now, but I've fixed the `load_*` parts for the contributor and recommendation feeds.

File: api/controller.py [hunk 3]
```diff
@@ -914,10 +926,15 @@ def related(self, data_source, identifier_type, identifier,
             return NO_SUCH_LANE.detailed(e.message)
 
         annotator = self.manager.annotator(lane)
+        if isinstance(facets, ProblemDetail):
```
Comment (aslagle) [inline api/controller.py]: do you need to load the facets for this one?

Comment (leonardr) [inline api/controller.py]: :blush:

**Comment by aslagle** [comment]
:+1:

**Pull request merged**